python3/bin/perfmon

```diff
@@ -467,13 +472,16 @@ def get_percent_mem_usage(ignored):
     try:
         with open("/proc/meminfo", "r", encoding="utf-8") as memfd:
             memlist = memfd.readlines()
-        memdict = [m.split(":", 1) for m in memlist]
-        memdict = dict(
-            [
-                (k.strip(), float(re.search(r"\d+", v.strip()).group(0)))
-                for (k, v) in memdict
-            ]
-        )
+        # memorylists is a list of lists, each list contains two parts: memtype and size
+        memorylists = [m.split(":", 1) for m in memlist]
+        memdict = {}
+        for item in memorylists:
+            memtype = item[0].strip()
+            size = item[1].strip()
+            match = re.search(r"\d+", size.strip())
+            if match is None:
+                raise NotGetValueException
```

Comment:
It would be helpful to code more error info (e.g. the line) in the exception. And the Exception would work well here.